os/arch/arm/src/imxrt/imxrt_mtd_flash.c

```diff
@@ -0,0 +1,676 @@
+	/* Save the FLASH geometry */
+
+	priv->sectorshift = IMXRT_1020EVK_NORFLASH_SECTOR_SHIFT;
+	priv->nsectors = IMXRT_1020EVK_NORFLASH_NSECTORS;
+	priv->pageshift = IMXRT_1020EVK_NORFLASH_PAGE_SHIFT;
+	priv->npages = IMXRT_1020EVK_NORFLASH_NPAGES;
+	#elif defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT105x)
+	status = imxrt_flexspi_nor_hyperflash_cfi(IMXRT_FLEXSPI);
+	/* Get vendor ID. */
+	if (status != kStatus_Success) {
+		return -ENODEV;
+	} else {
+		fvdbg("Found the HyperFlash by CFI\r\n");
+	}
+
+	/* Save the FLASH geometry */
+
+	priv->sectorshift = IMXRT_1050EVK_HYPERFLASH_SECTOR_SHIFT;
+	priv->nsectors = IMXRT_1050EVK_HYPERFLASH_NSECTORS;
+	priv->pageshift = IMXRT_1050EVK_HYPERFLASH_PAGE_SHIFT;
+	priv->npages = IMXRT_1050EVK_HYPERFLASH_NPAGES;
+	#endif
+	return OK;
+}
+
+/************************************************************************************
+ * Name: imxrt_waitwritecomplete
+ ************************************************************************************/
+
+static void imxrt_waitwritecomplete(struct imxrt_dev_s *priv)
+{
+	status_t status;
+
+	status = imxrt_flexspi_nor_wait_bus_busy(IMXRT_FLEXSPI);
+
+	if (status != kStatus_Success) {
+		fdbg("imxrt_flexspi_nor_wait_bus_busy: error[%d]\n", status);
+	}
+
+}
+
+/************************************************************************************
+ * Name:  imxrt_sectorerase
+ ************************************************************************************/
+
+static void imxrt_sectorerase(struct imxrt_dev_s *priv, off_t sector)
+{
+	status_t status;
+	off_t offset;
+
+	#ifdef CONFIG_IMXRT_SUBSECTOR_ERASE
+	if (priv->subsectorshift > 0) {
+		offset = sector << priv->subsectorshift;
+	} else
+	#endif
+	{
+		offset = sector << priv->sectorshift;
+	}
+
+	fdbg("priv: 0x%x,  IMXRT_FLEXSPI: 0x%x,  offset: 0x%x\n", priv, IMXRT_FLEXSPI, (long)offset);
+
+	status = imxrt_flexspi_nor_flash_erase_sector(IMXRT_FLEXSPI, offset);
+	if (status != kStatus_Success) {
+		fdbg("Erased fail!\n");
+		return;
+	}
+
+	fdbg("Erased\n");
+}
+
+/************************************************************************************
+ * Name:  imxrt_bulkerase
+ ************************************************************************************/
+
+static inline int imxrt_bulkerase(struct imxrt_dev_s *priv)
+{
+	fdbg("priv: %p\n", priv);
+
+    #ifdef CONSIDER_IMPLEMENT
+	/* Wait for any preceding write to complete.  We could simplify things by
+	 * perform this wait at the end of each write operation (rather than at
+	 * the beginning of ALL operations), but have the wait first will slightly
+	 * improve performance.
+	 */
+
+	imxrt_waitwritecomplete(priv);
+
+	/* Send write enable instruction */
+
+	imxrt_writeenable(priv);
+
+	/* Select this FLASH part */
+
+	SPI_SELECT(priv->dev, SPIDEV_FLASH, true);
+
+	/* Send the "Bulk Erase (BE)" instruction */
+
+	(void)SPI_SEND(priv->dev, IMXRT_BE);
+
+	/* Deselect the FLASH */
+
+	SPI_SELECT(priv->dev, SPIDEV_FLASH, false);
+	#endif
+
+	fdbg("Return: OK\n");
```

Comment:
Please remove unnecessary prints like this. 
If required, you can add debug logs with proper message.